README.md

```diff
@@ -449,7 +422,51 @@ FIXME: we should clarify this section and add mention of Samsung Browser as it's
    browser to a backend endpoint that
    redirects the browser to a URL with the intent
    scheme it is possible to use an existing AS
-   without modifications - though compared to App Links redirecting via the web browser is a poorer UX and leaves a tab open in the user's browser.
+   without modifications.
+
+   **Note:** Since the Intent scheme is handled as
+   a normal custom scheme outside of Android, it is
+   important that the endpoint that redirects the
+   user tries to determine whether the request came 
+   from an Android device or another device.
+   This is due to the fact that custom schemes introduce
+   security risks to the OAuth flow. If the
+   endpoint detects other devices it should display
+   an error message.
+
+
+### User's Default Browser Selection Considerations
+The selection of any browser that the user has set as the default 
+browser bears risks for user experience and security. 
+
+#### Overview
+
+**Google Chrome:**
+   - supports Android App Links, Intent schemes and custom schemes
+   - does not ask the user whether he wants to change the app
+
+**Mozilla Firefox:**
+   - has an option to open links in external apps (disabled by default)
+   - does not correctly validate Android App Links 
+   (relevant if the previous option is enabled)
+   - despite the external app option it always opens
+   Intent scheme in an app if there is no fallback url
+   provided (``S.browser_fallback_url=[encoded_full_url]`` SHOULD NOT 
+   be used in the Intent scheme)
+
+**Samsung Internet Browser:**
+   - has also an option to open links in external apps (disabled by default)
+   - does correctly open Intent schemes even if a fallback
+   url is provided (without enabling the previous option)
+   - handles Android App Links correctly if the first option
+   is enabled
+
+**DuckDuckGo and Puffin:** 
+   - if the browser is redirected to a URL
+   with a custom scheme that opens another app, the
+   browser warns the user.
```

Comment:
is the implication here that these two browsers don't support intent: urls nor app links, and custom schemes is the only way that can be used? If so perhaps we should say that explicitly.

Reply:
No these browsers support the intent:// scheme but they always ask the user if he really wants to jump from the browser to the app. This can be seen in the image below.

Reply:
Should it have a line the same as chrome does then, "`- supports Android App Links, Intent schemes and custom schemes`"? It just seems odd that duckduckgo/puffin we only mention custom schemes, whereas the other 3 we explicitly mention applinks/intent scheme.